modules/geoedgeRtdProvider.js

```diff
@@ -0,0 +1,238 @@
+}
+
+/**
+ * sets the wrapper and calls preload client
+ * @param {string} responseText
+ */
+export function setWrapper(responseText) {
+  wrapperReady = true;
+  wrapper = responseText;
+}
+
+/**
+ * preloads the client
+  * @param {string} key
+ */
+export function preloadClient(key) {
+  let link = document.createElement('link');
+  link.rel = 'preload';
+  link.as = 'script';
+  link.href = getClientUrl(key);
+  link.onload = () => { preloaded = true };
+  insertElement(link);
+}
+
+/**
+ * creates identity function for string replace without special replacement patterns
+ * @param {string} str
+ * @return {function}
+ */
+function replacer(str) {
+  return function () {
+    return str;
+  }
+}
+
+export function wrapHtml(wrapper, html) {
+  return wrapper.replace(HTML_PLACEHOLDER, replacer(html));
+}
+
+/**
+ * generate macros dictionary from bid response
+ * @param {Object} bid
+ * @param {string} key
+ * @return {Object}
+ */
+function getMacros(bid, key = providerParams.key) {
+  return {
+    '${key}': key,
+    '%%ADUNIT%%': bid.adUnitCode,
+    '%%WIDTH%%': bid.width,
+    '%%HEIGHT%%': bid.height,
+    '%%PATTERN:hb_adid%%': bid.adId,
+    '%%PATTERN:hb_bidder%%': bid.bidderCode,
+    '%_isHb!': true,
+    '%_hbcid!': bid.creativeId || '',
+    '%%PATTERN:hb_pb%%': bid.pbHg,
+    '%%SITE%%': location.hostname,
+    '%_pimp%': PV_ID
+  };
+}
+
+/**
+ * replace macro placeholders in a string with values from a dictionary
+ * @param {string} wrapper
+ * @param {Object} macros
+ * @return {string}
+ */
+function replaceMacros(wrapper, macros) {
+  let key;
+  for (key in macros) {
+    wrapper = wrapper.replace(key, replacer(macros[key]));
+  }
+  return wrapper;
+}
+
+/**
+ * build final creative html with creative wrapper
+ * @param {Object} bid
+ * @param {string} wrapper
+ * @param {string} html
+ * @return {string}
+ */
+function buildHtml(bid, wrapper, html, key) {
+  let macros = getMacros(bid, key);
+  wrapper = replaceMacros(wrapper, macros);
+  return wrapHtml(wrapper, html);
+}
+
+/**
+ * muatates the bid ad property
+ * @param {Object} bid
+ * @param {string} ad
+ */
+function mutateBid(bid, ad) {
+  bid.ad = ad;
+}
+
+/**
+ * wraps a bid object with the creative wrapper
+ * @param {Object} bid
+ * @param {string} key
+ */
+export function wrapBidResponse(bid, key = providerParams.key) {
+  let wrapped = buildHtml(bid, wrapper, bid.ad, key);
+  mutateBid(bid, wrapped);
+}
+
+/**
+ * checks if bidder's bids should be monitored
+ * @param {string} bidder
+ * @return {boolean}
+ */
+function isSupportedBidder(bidder) {
+  return isEmpty(providerParams.bidders) || providerParams.bidders[bidder] === true;
+}
+
+/**
+ * checks if bid should be monitored
+ * @param {Object} bid
+ * @return {boolean}
+ */
+function shouldWrap(bid) {
+  let supportedBidder = isSupportedBidder(bid.bidderCode);
+  let wap = providerParams.wap ? preloaded : true;
+  return wrapperReady && supportedBidder && wap;
+}
+
+function conditionallyWrap(bid) {
```

Comment:
Incorrect. This will be called with 3 params bidResponse, config, userConsent. Get your 'params' from there instead of calling getConfig directly.